### PR TITLE
[TS] LPP-46273 > LPS-161050 Keep track on the fieldset's fields on rendering

### PR DESCRIPTION
**Pull request by hongvo2308**
Hi,
This is a PR for fixing the issue in the ticket https://issues.liferay.com/browse/LPP-46273
Please review this and leave your comments.
Thank you.

### CODE REVIEW

**Comment by hongvo2308** [comment]
ci:test:bundle

**Comment by rodrigopaulino** [comment]
ci:test:form

**Comment by rodrigopaulino** [comment]
ci:test:data-engine

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
Hi, @hongvo2308. Thanks for contributing.

Could you please use `findField` functions instead? It's better in performance because it will short circuit, once it finds a field that matches the condition.

Thanks.

**Comment by hongvo2308** [comment]
Hi @rodrigopaulino 
Thank you for reviewing this PR and suggesting a better solution. I have updated this PR based on your suggestion. Could you please take a look at it again?
Thank you.

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/PageRenderer/Page.es.js [hunk 1]
```diff
@@ -176,10 +176,8 @@ const Page = ({
 	if (pages?.[activePage]) {
 		const visitor = new PagesVisitor([pages[activePage]]);
 
-		visitor.mapFields((field) => {
-			if (field.required) {
-				hasFieldRequired = true;
-			}
+		hasFieldRequired = visitor.findField((field) => {
+			return field.required;
 		});
```
Comment (rodrigopaulino) [inline modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/PageRenderer/Page.es.js:181]: Good job, @hongvo2308. I'd like to suggest just a couple of tweaks.
Can you test that without the `return` statement, also `hasFieldRequired` is supposed to be a boolean, so you can convert the found field into a boolean like this:

```suggestion
		hasFieldRequired = !!visitor.findField((field) => field.required);
```
Thx

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]

**Comment by rodrigopaulino** [comment]
After you're done, @hongvo2308. Can you run the suites sf and relevant? Thx!

**Comment by hongvo2308** [comment]
Hi @rodrigopaulino 
Thank you for your suggestion. I have tested it on my local and it works properly. So I have updated this PR based on your suggestion.
Thank you.

**Comment by hongvo2308** [comment]
ci:test:sf

**Comment by hongvo2308** [comment]
ci:test:relevant

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Comment by rodrigopaulino** [comment]
ci:forward

**Pull request closed without merging**